subprojects/core/src/main/java/org/gradle/api/internal/project/BuildOperationCrossProjectConfigurator.java

```diff
@@ -30,12 +31,7 @@
 public class BuildOperationCrossProjectConfigurator implements CrossProjectConfigurator {
 
     private final BuildOperationExecutor buildOperationExecutor;
-    private final ThreadLocal<Boolean> allowExecution = new ThreadLocal<Boolean>() {
-        @Override
-        protected Boolean initialValue() {
-            return true;
-        }
-    };
+    private final MutationGuard mutationGuard = new DefaultMutationGuard();
```

Comment:
❓ I'm wondering if we can move this `MutationGuard` instance inside `DefaultGradle` instance and just have all `Project` fetch the guard from there and below, we could use `MutationGuards.of(project)` to get the guard. This assumes a single `Gradle` instance is constructed as `CrossProjectConfigurer` is instantiated once per build session.